lib/resources/shadow.rb

```diff
@@ -121,10 +120,14 @@ def to_s
       "/etc/shadow#{f}"
     end
 
+    def params
+      @params = @lines.nil? ? [] : @lines.map { |l| parse_shadow_line(l) }
+    end
```

Comment:
Couple of directions this could go to make the point more obvious.

You could separate the guard statement:
```ruby
def params(lines)
  return [] if lines.nil?

  lines.map { |l| parse_shadow_line(1) }
end
```
And that maps directly to your logic in the ternary. But what I think you're actually concerned about is calling `map` on `nil`? So this would actually be more direct and clear:
```ruby
def params(lines)
  Array(lines).map { |l| parse_shadow_line(1) }
end
```
`Array(nil)` will produce the empty array, and pass through the `map` call to return what you expect, just that empty array. So there's one path through the code and all the needed safety.